geonode_mapstore_client/client/js/utils/AppRoutesUtils.js

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2022, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { ResourceTypes } from '@js/utils/ResourceUtils';
+
+export const appRouteComponentTypes = {
+    resourceViewer: 'ViewerRoute',
+    searchComponent: 'SearchRoute',
+    detailComponent: 'DetailRoute',
+    datasetUploadComponent: 'UploadDatasetRoute',
+    documentUploadComponent: 'UploadDocumentRoute',
+    homeComponent: 'Home'
+};
```

Comment:
Please update name of these constants accordingly in other files
Note: that I have also changed 'Home' to 'HomeRoute'

```suggestion
export const AppRouteComponentTypes = {
    VIEWER: 'ViewerRoute',
    SEARCH: 'SearchRoute',
    DETAIL: 'DetailRoute',
    DATASET_UPLOAD: 'UploadDatasetRoute',
    DOCUMENT_UPLOAD: 'UploadDocumentRoute',
    HOME: 'HomeRoute'
};
```